### DISCUSSION TITLE
How to enable autopep8

### DISCUSSION THREAD

**Discussion by kresdjan** [emacs-lsp/lsp-mode#3223] [Q&A] [answered]
I have installed `pylsp` using `python3 -m pip install 'python-lsp-server[all]'`.
I can confirm autopep8 is installed via the Terminal `$ autopep8 test.py`.

But I can't get it working in emacs. I have just naively set `(setq lsp-pyls-plugins-autopep8-enabled t)` to True. When trying to format an incorrectly formatted file `M-x lsp-format-buffer` it returns `LSP :: No formatting changes provided`.

How do I get `autopep8` working with emacs-lsp?

Thanks.

**Comment by yyoncho** [comment]
@mpanarin might know.

**Comment by mpanarin** [comment, marked as answer]
@kresdjan It seems that you are setting a variable for the wrong server.
You need `(setq lsp-pylsp-plugins-autopep8-enabled t)`

note the `pylsp` instead of `pyls` in the variable name

**Reply by kresdjan** [reply]
@mpanarin how embarrassing. I've just swapped it out for `(setq lsp-pylsp-plugins-yapf-enabled t)`. Removed `autopep8` and installed `yap` still seeing `No formatting changes provided`. I can only run it from the Terminal `yapf -i file.py`.
